hedera-node/src/test/java/com/hedera/services/contracts/execution/CallEvmTxProcessorTest.java

```diff
@@ -342,6 +369,397 @@ void assertTransactionSenderAndValue() {
 		assertEquals(transaction.getValue(), buildMessageFrame.getApparentValue());
 	}
 
+	@Test
+	void assertSuccessEthereumTransactionExecutionChargesBothSenderAndRelayerWithoutRefunds() {
+		givenValidMockEth();
+		given(merkleNetworkContextSupplier.get()).willReturn(merkleNetworkContext);
+		final var MAX_REFUND_PERCENTAGE = 100;
+		given(globalDynamicProperties.maxGasRefundPercentage()).willReturn(MAX_REFUND_PERCENTAGE);
+		given(globalDynamicProperties.fundingAccount()).willReturn(new Id(0, 0, 1010).asGrpcAccount());
+		given(storageExpiry.hapiCallOracle()).willReturn(oracle);
+		final var wrappedSenderAccount = mock(EvmAccount.class);
+		final var mutableSenderAccount = mock(MutableAccount.class);
+		given(wrappedSenderAccount.getMutable()).willReturn(mutableSenderAccount);
+		given(updater.getOrCreateSenderAccount(sender.getId().asEvmAddress())).willReturn(wrappedSenderAccount);
+		given(mutableSenderAccount.getBalance()).willReturn(Wei.of(100* ONE_HBAR));
+		final var wrappedRelayerAccount = mock(EvmAccount.class);
+		final var mutableRelayerAccount = mock(MutableAccount.class);
+		given(wrappedRelayerAccount.getMutable()).willReturn(mutableRelayerAccount);
+		given(updater.getOrCreateSenderAccount(relayer.getId().asEvmAddress())).willReturn(wrappedRelayerAccount);
+		given(mutableRelayerAccount.getBalance()).willReturn(Wei.of(100* ONE_HBAR));
+		final long gasPrice = 40L;
+		given(livePricesSource.currentGasPrice(consensusTime, HederaFunctionality.EthereumTransaction))
+				.willReturn(gasPrice);
+		final var receiverAddress = receiver.getId().asEvmAddress();
+		given(aliasManager.resolveForEvm(receiverAddress)).willReturn(receiverAddress);
+		final long offeredGasPrice = 10L;
+		final int gasLimit = 1000;
+		given(gasCalculator.transactionIntrinsicGasCost(Bytes.EMPTY, false)).willReturn(Gas.of(gasLimit));
+
+		var result = callEvmTxProcessor.executeEth(
+				sender, receiverAddress, gasLimit, 1234L, Bytes.EMPTY, consensusTime,
+				BigInteger.valueOf(offeredGasPrice).multiply(WEIBARS_TO_TINYBARS),
+				relayer, 10 * ONE_HBAR);
+
+		assertTrue(result.isSuccessful());
+		assertEquals(result.getGasUsed(), gasLimit);
+		assertEquals(receiver.getId().asGrpcContract(), result.toGrpc().getContractID());
+		verify(mutableRelayerAccount,
+				never()).incrementBalance(any());
+		verify(mutableSenderAccount, never()).incrementBalance(any());
```

Comment:
Missing a `verify(mutableRelayerAccount).decrementBalance(...))` and a `verify(mutableSenderAccount).decrementBalance(...))`

Reply:
Same for `assertSuccessEthereumTransactionExecutionChargesBothSenderAndRelayerAndRefunds` and `assertSuccessEthereumTransactionExecutionChargesBothSenderAndRelayerAndRefundsOnlyRelayer`